py-polars/tests/unit/operations/rolling/test_rolling.py

```diff
@@ -955,3 +955,44 @@ def test_rolling_invalid() -> None:
             .rolling("index", period="3000d")
             .agg(pl.col("values").sum().alias("sum"))
         )
+
+
+@pytest.mark.parametrize("time_unit", ["ms", "us", "ns"])
+def test_rolling_duration(time_unit):
+    # Note: Both datetime with Unit != ns and Duration have weird behavior.
```

Comment:
could you clarify what you mean here please?

Reply:
It's in the issue a bit lower in #14764. Although I haven't had time to investigate the current implementation for `DateTime` period over `Int64` index columns sufficiently to make a well-written issue, the current behavior of always assuming the index column represents nanoseconds seems a little surprising. I'd like to separate that from this PR though, since it is a different issue, as you stated.